Circular_kernel_3/include/CGAL/Circular_kernel_3/Circular_arc_3.h

```diff
@@ -282,7 +282,15 @@ namespace CGAL {
         const double dz = z2-z1;
         const double d_sq = dx*dx + dy*dy + dz*dz;
         const double r_sq = to_double(squared_radius());
-        const double ap_ang = 2.0 * std::asin(0.5 * std::sqrt(d_sq / r_sq));
+        const double s = 0.5 * std::sqrt(d_sq / r_sq);
+        double ap_ang;
+        if(std::abs(s)<=1) {
+          ap_ang = 2.0 * std::asin(s);
+        } else {
+          // We only allow a small rounding error
+          CGAL_assertion(std::abs(s)<=1.0001);
+          ap_ang = (s < 0) ? -CGAL_PI : CGAL_PI;
+        }
```

Comment:
I do not see where this maps on a call with 3 points.   It would if there was the center of the circle of the arc.

Reply:
Actually, this function takes one circle of radius `r`, two points `P`, and `Q`,  on it, and compute the angle formed by the arc between the two points by using the formula `2×asin(|PQ|/(2×r))`. That is different, because it does not need to know the center of the circle, but only its radius.